pkgs/by-name/sf/sfml/package.nix

```
@@ -0,0 +1,93 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  libX11,
+  freetype,
+  libjpeg,
+  openal,
+  flac,
+  libvorbis,
+  glew,
+  libXrandr,
+  libXrender,
+  udev,
+  xcbutilimage,
+  darwin,
+  libXcursor,
+  fetchpatch,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "sfml";
+  version = "2.6.1";
+
+  src = fetchFromGitHub {
+    owner = "SFML";
+    repo = "SFML";
+    rev = finalAttrs.version;
+    hash = "sha256-R+ULgaKSPadcPNW4D2/jlxMKHc1L9e4FprgqLRuyZk4=";
+  };
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs =
+    [
+      freetype
+      libjpeg
+      openal
+      flac
+      libvorbis
+      glew
+    ]
+    ++ lib.optional stdenv.isLinux udev
+    ++ lib.optionals (!stdenv.isDarwin) [
+      libX11
+      libXrandr
+      libXrender
+      xcbutilimage
+      libXcursor
+    ]
+    ++ lib.optionals stdenv.isDarwin (
+      with darwin.apple_sdk.frameworks;
+      [
+        IOKit
+        Foundation
+        AppKit
+        OpenAL
+      ]
+    );
+
+  cmakeFlags = [
+    "-DSFML_INSTALL_PKGCONFIG_FILES=yes"
+    "-DSFML_MISC_INSTALL_PREFIX=share/SFML"
+    "-DSFML_BUILD_FRAMEWORKS=no"
+    "-DSFML_USE_SYSTEM_DEPS=yes"
+  ];
+
+  patches = [
+    # Fix pkg-config
+    # See https://github.com/SFML/SFML/issues/2815
+    # Also, too much changes in CMakeLists.txt and changelog.md,
+    # so we patchin cmake ourself
+    (fetchpatch {
+      url = "https://patch-diff.githubusercontent.com/raw/SFML/SFML/pull/2835.patch";
+      hash = "sha256-kdOAXR9YPQllx64z9dgwCV+vy0cJvIsZZboZKFc4Q8Q=";
+      excludes = [ "changelog.md" "CMakeLists.txt" ];
+    })
+    ./CMakeLists.txt-pkgconfig.patch
+  ];
+
```

Lets split the package between `out` & `dev`
```suggestion

  outputs = [ "out" "dev" ];
```